src/server/svr_func.c

```diff
@@ -4717,6 +4720,30 @@ is_runnable(job *ptr, struct prov_vnode_info *pvnfo)
 				break;
 			}
 		}
+		if (((mom_svrinfo_t *)(np->nd_moms[0]->mi_data))->msr_is_hook_checksums_done)
+			 /* count nodes who have processed IS_HOOK_CHECKSUMS message */
+			syncnodes++;
+	}
+	if (!eflag 
+		&& (syncnodes != num_of_prov_vnodes
+		|| do_sync_mom_hookfiles
+		|| sync_mom_hookfiles_proc_running)) {
+
+		/* check if IS_HOOK_CHECKSUMS is processed if not then
+		 * come back after 5 sec and check again, this affirms
+		 * the hook files are in sync before starting the job 
+		 **/
+
+		DBPRT(("%s: node:%s not yet processed IS_HOOK_CHECKSUMS\n",
+			__func__, pvnfo->pvnfo_vnode))
+
+		/* set a work task to come back here after 5 sec */ 
+		if (!set_task(WORK_Timed, time_now + 5, 
+			is_sync_hooks_done, pvnfo)){
```

Comment:
In that case, it has to traverse the tree to fetch node information hence keeping as it is.